File: leetcode/004/java/Solution.java
/**
 * ---------------------------------------------------------------------------------------
 *
 * Title:       Median of Two Sorted Arrays
 *
 * Link:        https://leetcode.com/problems/median-of-two-sorted-arrays/
 *
 * Difficulty:  Hard
 *
 * Language:    Java
 *
 * ---------------------------------------------------------------------------------------
 */

class Solution {

    public static void main(String[] args) {
        (new Solution()).go();
    }

    public void go() {
        int[] nums1 = new int[] { 1,4,7,8,10,14 };
        int[] nums2 = new int[] { 3,6,8,55 };
        System.out.println(findMedianSortedArrays(nums1,nums2));
    }

    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int[] x = new int[nums1.length+nums2.length];
        int i=0, j=0, k=0;

        while(true) {
            boolean done = false;
            if( i < nums1.length && ( j >= nums2.length || nums1[i] <= nums2[j] ) ) {
                x[k] = nums1[i];
                i++;
                k++;
                done = true;
            }

            if ( !done && j < nums2.length && ( i >= nums1.length || nums2[j] <= nums1[i] ) ) {
                x[k] = nums2[j];
                j++;
                k++;
                done = true;
            }
            if( !done )
                break;
        }

        double rv;
        i = x.length / 2;

        if( x.length % 2 == 0 )
            rv = (double)( x[i] + x[i-1] ) / 2.0;
        else
            rv = (double) x[i];

        return rv;
    }

}

File: leetcode/004/python/solution.py
# ---------------------------------------------------------------------------------------
#
#  Title:       Median of Two Sorted Arrays
#
#  Link:        https://leetcode.com/problems/median-of-two-sorted-arrays/
#
#  Difficulty:  Hard
#
#  Language:    Python
#
# ---------------------------------------------------------------------------------------

class Solution(object):
    def findMedianSortedArrays(self, nums1, nums2):
        i = 0
        j = 0
        x = []
        while(True):
            done = False
            if i < len(nums1) and ( j >= len(nums2) or nums1[i] <= nums2[j] ):
                x.append(nums1[i])
                i += 1
                done = True
            if not done and j < len(nums2) and ( i >= len(nums1) or nums2[j] <= nums1[i] ):
                x.append(nums2[j])
                j += 1
                done = True
            if not done:
                break

        lenx = len(x)
        midx = lenx >> 1
        rv = 0.0
        if lenx % 2 == 0:
            rv = (x[midx] + x[midx-1]) / 2.0
        else:
            rv = x[midx]
        return rv

if __name__ == "__main__":
    z = Solution()
    print(z.findMedianSortedArrays([1,8],[3,4,7,9]))
